pkg/controller/install/add_openshift.go

```diff
@@ -5,6 +5,7 @@ import (
 )
 
 func init() {
-	platformPrereqFuncs = append(platformPrereqFuncs, openshift.EnsureMaistra)
+	platformPreInstallFuncs = append(platformPreInstallFuncs, openshift.EnsureMaistra)
```

Comment:
I'm glad you changed that name :)

Reply:
But since k8s is "eventually consistent" anyway, is there any compelling need to break functions into pre and post install? Just like the k-s operator eventually sees the istio CRD it needs (and keeps trying if it doesn't) because the maistra operator is so slow, what's the harm in having the ingress operator go pre-install? If not, let's collapse pre and post into a single "extra" funcs list (or some better name).

Reply:
The harm is just in avoiding a restart of the ingress operator if it comes up before the Knative CRDs get installed, I guess. We can try it pre-install.

However, I believe things like https://github.com/openshift-cloud-functions/knative-operators/blob/90c37133e794ac5cd6de2648181c2f1f8d1190aa/etc/scripts/installation-functions.sh#L307-L322 will have to happen post-install as they munge resources created from the install itself.